dataactvalidator/validation_handlers/validator.py

```diff
@@ -168,13 +168,24 @@ def cross_validate_sql(rules, submission_id, short_to_long_dict):
             cols = failed_rows.keys()
             cols.remove('row_number')
             column_string = ", ".join(short_to_long_dict[c] if c in short_to_long_dict else c for c in cols)
+
+            # materialize as we'll iterate over the failed_rows twice
+            failed_rows = list(failed_rows)
+            flex_data = relevant_cross_flex_data(failed_rows, submission_id, [first_file, second_file])
+
             for row in failed_rows:
                 # get list of values for each column
                 values = ["{}: {}".format(short_to_long_dict[c], str(row[c])) if c in short_to_long_dict else
                           "{}: {}".format(c, str(row[c])) for c in cols]
                 values = ", ".join(values)
+                full_column_string = column_string
+                # go through all flex fields in this row and add to the columns and values
+                for field in flex_data[row['row_number']]:
+                    full_column_string += ", " + field.header + "_file" + field.file_letter_name.lower()
```

Comment:
I'm not sure if we check for duplicates. If we do, that's not what the issue is here. It's possible to have a flex field called `flex_divOp` in file A and also have one called `flex_divOp` in file B and that's not disallowed because it might be needed. We need to differentiate between the two somehow in the error/warning files for cross-file validations since columns from both files are in the output files